server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -242,7 +250,11 @@ private void dispatchRequest(RestRequest request, RestChannel channel, RestHandl
                 inFlightRequestsBreaker(circuitBreakerService).addWithoutBreaking(contentLength);
             }
             // iff we could reserve bytes for the request we need to send the response also over this channel
-            responseChannel = new ResourceHandlingHttpChannel(channel, circuitBreakerService, contentLength);
+            // Using a version from a handler because if no handler was found for requested version,
```

Comment:
I am not sure i understand the purpose of this comment here.

Reply:
my intention was to indicate that even if you requested an `applicaiton/vnd.elasticsearch+json;compatible-with=7` but the handler did not exist in 7, the actual resthandler will be used from v8.
but that indeed makes no difference, as there would be no compatible serialisation code.
Removed the comment.